akka-projection-r2dbc-integration/src/test/scala/akka/projection/r2dbc/EventSourcedEndToEndSpec.scala

```diff
@@ -214,77 +344,120 @@ class EventSourcedEndToEndSpec
+          }
       }
     }
   }
 
-  s"A R2DBC projection with eventsBySlices source (dialect ${r2dbcSettings.dialectName})" must {
+  private def test(
+      startParams: StartParams,
+      startProjectionsFactory: () => Vector[ActorRef[ProjectionBehavior.Command]]): Unit = {
+    val numberOfEntities = 20 // increase this for longer testing
+    val numberOfEvents = numberOfEntities * 10
 
-    "handle all events exactlyOnce" in {
-      val numberOfEntities = 20
-      val numberOfEvents = numberOfEntities * 10
-      val entityType = nextEntityType()
+    val entities = (0 until numberOfEntities).map { n =>
+      val persistenceId = PersistenceId(startParams.entityType, s"p$n")
+      spawn(Persister(persistenceId), s"${startParams.entityType}-p$n")
+    }
 
-      val entities = (0 until numberOfEntities).map { n =>
-        val persistenceId = PersistenceId(entityType, s"p$n")
-        spawn(Persister(persistenceId), s"$entityType-p$n")
+    // write some before starting the projections
+    var n = 1
+    while (n <= numberOfEvents / 4) {
+      val p = n % numberOfEntities
+      // mix some persist 1 and persist 3 events
+      if (n % 7 == 0) {
+        entities(p) ! Persister.PersistAll((0 until 3).map(i => mkEvent(n + i)).toList)
+        n += 3
+      } else {
+        entities(p) ! Persister.Persist(mkEvent(n))
+        n += 1
       }
 
-      // write some before starting the projections
-      var n = 1
-      while (n <= 50) {
-        val p = n % numberOfEntities
-        // mix some persist 1 and persist 3 events
-        if (n % 7 == 0) {
-          entities(p) ! Persister.PersistAll((0 until 3).map(i => mkEvent(n + i)).toList)
-          n += 3
-        } else {
-          entities(p) ! Persister.Persist(mkEvent(n))
-          n += 1
+      if (n % 10 == 0)
+        Thread.sleep(50)
+      else if (n % 25 == 0)
+        Thread.sleep(1500)
+    }
+
+    val projections = startProjectionsFactory()
+
+    // give them some time to start before writing more events
+    Thread.sleep(500)
+
+    while (n <= numberOfEvents) {
+      val p = n % numberOfEntities
+      entities(p) ! Persister.Persist(mkEvent(n))
+
+      // stop projections
+      if (n == numberOfEvents / 2) {
+        projections.foreach { ref =>
+          ref ! ProjectionBehavior.Stop
         }
       }
 
-      val projectionName = UUID.randomUUID().toString
-      val processedProbe = createTestProbe[Processed]()
-      var projections = startProjections(entityType, projectionName, nrOfProjections = 4, processedProbe.ref)
+      // wait until stopped
+      if (n == (numberOfEvents / 2) + 10) {
+        val probe = createTestProbe()
+        projections.foreach { ref =>
+          probe.expectTerminated(ref)
+        }
+      }
 
-      // give them some time to start before writing more events
-      Thread.sleep(500)
+      // resume projections again
+      if (n == (numberOfEvents / 2) + 20)
+        startProjectionsFactory()
```

Comment:
I changed to `var projections` recently. This would be `projections = startProjectionsFactory()`.
The reason is that later we are stopping the projections, but that is not these new projections.

Reply:
Ah, missed that when aligning with the dynamodb test. I'll fix both.